src/nupic/os/Regex.cpp

```diff
@@ -46,7 +47,14 @@ namespace nupic
       exactRegExp += re;
       if (re[re.length()-1] != '$') 
         exactRegExp += '$';
-      
+
+#if defined(NTA_OS_WINDOWS)
```

Comment:
This PR still needs either 1) to switch back to PCRE for all builds (preferred) or 2) a comment with the issue link for moving over to std::regex